examples/units_and_coordinates/radec_to_hpc_map.py

```diff
@@ -0,0 +1,134 @@
+"""
+================
+Create a Helioprojective Map from observations in the RA-DEC coordinate system
+================
+
+How to create a `~sunpy.map.Map` in Helioprojective Coordinate Frame from radio observations
+in ICRS (RA-DEC).
+
+In this example a LOFAR FITS file (make from CASA) is read in and the WCS
+header information is then used to make a new header with the information in
+Helioprojective and a `~sunpy.map.Map` is made.
+
+The LOFAR example file has a WCS in celestial coordinates i.e. Right Ascension and
+Declination (RA-DEC). For many solar studies we may want to plot this data in some
+Sun-centered coordinate frame, such as Helioprojective. In this example we read the
+data and header information from the LOFAR FITS file and then create a new header
+with updated WCS information to create a `~sunpy.map.Map` with a HPC coordinate frame.
+We will make use of the `astropy.coordinates` and `sunpy.coordinates` submodules
+together with `sunpy.map.make_fitswcs_header` to create a new header and generate
+a `sunpy.map.Map`.
+"""
+import matplotlib.pyplot as plt
+import numpy as np
+
+from astropy import units as u
+from astropy.coordinates import EarthLocation, SkyCoord
+from astropy.io import fits
+from astropy.time import Time
+
+import sunpy.data.sample
+import sunpy.map
+from sunpy.coordinates import frames, sun
+
+##############################################################################
+# We will first begin be reading in the header and data from the FITS file.
+hdu = fits.open(sunpy.data.sample.LOFAR_IMAGE)
+header = hdu[0].header
+data = np.squeeze(hdu[0].data)
+
+###############################################################################
+# We can inspect the header, for example, we can print the coordinate system and
+# type projection, which here is RA-DEC.
+print(header['ctype1'], header['ctype2'])
+
+###############################################################################
+# Lets pull out the observation time and wavelength from the header, we will use
+# these to create our new header.
+obstime = Time(header['date-obs'])
+frequency = header['crval3']*u.Hz
+
+###############################################################################
+# To create a new `~sunpy.map.Map` header we need convert the reference coordinate
+# in RA-DEC to Helioprojective. To do this we will first create an `astropy.coordinates.SkyCoord`
+# of the reference coordinate from the header information. LOFAR observation are in
+# the J2000 epoch (this information is in the header).
+reference_coord = SkyCoord(header['crval1']*u.deg, header['crval2']*u.deg,
+                           frame='gcrs',
+                           obstime=obstime,
+                           distance=sun.earth_distance(obstime),
+                           equinox='J2000')
```

Comment:
Questions/comments:

- Are you sure that the coordinates provided are in GCRS, as opposed to the ICRF-based frame of ICRS with a shifted origin?  The critical difference is whether to include stellar aberration due to observer motion (which shifts coordinates by ~20 arcseconds).
- If using GCRS is in fact correct, you should use the `GCRS` keyword arguments `obsgeoloc` and `obsgeovel` to precisely specify the observer location.
- The argument `equinox='J2000'` doesn't actually do anything here.  It's not used by `GCRS`, nor by any frame in the transformation path to `Helioprojective`.

Reply:
Incidentally, since you specifically note that LOFAR say that their coordinates are at the reference epoch of J2000.0, they may not be using an ICRF-based frame like ICRS or GCRS, which don't really have a reference epoch.  They could instead be using an equatorial coordinate system with a reference epoch (e..g, Astropy's `PrecessedGeocentric` for the mean equatorial coordinate system, which even at J2000.0 isn't exactly the same as GCRS).

Aren't coordinates fun?

Reply:
good question - so as far as I can tell it should be GCRS (this is mainly based on testing with an eclipse observation as we know exactly where the moon should be).

I also reached out to some people at ASTRON and they think its also in gcrs with comparisons with NRH observations, but they are going to look into it some more

Reply:
also the more I seem to learn about coordinates the more i'm getting confused

....ignorance was bliss

Reply:
So yes, according to them it should be in gcrs apparently (this example gives same results compared to their own codes)